src/marqo/core/unstructured_vespa_index/unstructured_document.py

```
@@ -0,0 +1,180 @@
+
+    match_features: Dict[str, Any] = Field(default_factory=dict, alias=unstructured_common.VESPA_DOC_MATCH_FEATURES)
+
+    # Fields that are excluded when generating vespa documents
+    _EXCLUDED_FIELDS = {"match_features"}
+
+    class Config:
+        allow_population_by_field_name = True
+
+    def to_vespa_dictionary(self) -> Dict[str, Any]:
+        # Exclude None and empty fields but keep 0 as we need to have vector_counts in docs
+        return {k: v for k, v in self.dict(exclude_none=True, by_alias=True,
+                                           exclude=self._EXCLUDED_FIELDS).items() if v or v == 0}
+
+    def to_marqo_doc(self, return_highlights: bool = False) -> Dict[str, Any]:
+        marqo_document = {}
+        # Processing short_string_fields and long_string_fields back into original format
+        marqo_document.update(self.short_string_fields)
+        marqo_document.update(self.long_string_fields)
+        # Reconstruct string arrays
+        for string_array in self.string_arrays:
+            key, value = string_array.split("::", 2)
+            if key not in marqo_document:
+                marqo_document[key] = []
+            marqo_document[key].append(value)
+
+        # Add int and float fields back
+        marqo_document.update(self.int_fields)
+        marqo_document.update(self.float_fields)
+        marqo_document.update({k: bool(v) for k, v in self.bool_fields.items()})
+        marqo_document[index_constants.MARQO_DOC_ID] = self.marqo__id
+
+        # vespa_embeddings may not return if it comes from a search result
+        embeddings_list = []
+        if self.vespa_embeddings:
+            try:
+                embeddings_list = list(self.vespa_embeddings["blocks"].values())
+            except (KeyError, AttributeError, TypeError):
+                raise VespaDocumentParsingError(f"No embeddings found in the document _id = {self.marqo__id}")
```

I wonder if the error message is accurate. Under what circumstances do you expect to reach here?

---

I think it should not reach here unless the data from vespa is broken, maybe raise an internal error?